src/js/jsx/shared/Datalist.jsx

```diff
@@ -314,6 +316,8 @@ define(function (require, exports, module) {
                 case "Tab":
                     return;
                 case "Enter":
+                    this.props.onKeyDown(event);
```

Comment:
I am not sure, I didn't even notice that. It also happens in `SearchBar.jsx` _handleKeyDown(). Removing the `Return` case from both.